modules/standard/GPU.chpl

```diff
@@ -132,4 +134,38 @@ module GPU
   proc gpuClocksPerSec(devNum : int) {
     return chpl_gpu_device_clock_rate(devNum : int(32));
   }
+
+  pragma "no doc"
+  type GpuAsyncCommHandle = c_void_ptr;
+
+  /*
+    Copy srcArr to dstArr, atleast one array must be on a GPU; this function
+    can be used for either communication to or from the GPU
+
+    Returns a handle that can be passed to `waitGpuComm` to pause execution
+    until completion of this asyhcronous transfer
+  */
+  pragma "no doc"
+  proc asyncGpuComm(dstArr, srcArr) : GpuAsyncCommHandle {
```

Comment:
Even though this is not supposed to be user-facing right now, the generic formals worry me. It looks like you can call this with two strings, and more scarily a string and an a data structure for example. I recommend:

```chapel
proc asyncGpuComm(dstArr: ?t, srcArr: t) where isArrayType(t): GpuAsyncCommHandle
```